cdi/src/main/java/brave/webmvc/TracingProducer.java

```diff
@@ -0,0 +1,109 @@
+package brave.webmvc;
+
+import static brave.webmvc.ConfigKeys.TRACING_REPORTER_URL;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Disposes;
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.CDI;
+
+import brave.Tracing;
+import brave.context.log4j2.ThreadContextScopeDecorator;
+import brave.propagation.B3Propagation;
+import brave.propagation.ThreadLocalCurrentTraceContext;
+import brave.sampler.Sampler;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import zipkin2.Span;
+import zipkin2.codec.Encoding;
+import zipkin2.reporter.AsyncReporter;
+import zipkin2.reporter.Reporter;
+import zipkin2.reporter.okhttp3.OkHttpSender;
+
+@ApplicationScoped
+class TracingProducer {
+
+    private static final Logger log = Logger.getLogger(TracingProducer.class.getName());
+
+    /**
+     * @return configured brave tracing
+     */
+    public static Tracing createTracing() {
+        return CDI.current().select(Tracing.class).get();
+    }
+
+    /**
+     * {@link Dependent} scope is used. {@link ApplicationScoped} does not work because
+     * {@link brave.Tracing#clock(brave.propagation.TraceContext)} is a {@code final} method.
+     *
+     * @param spanReporter injected reporter
+     * @param enabled value of {@link ConfigKeys#TRACING_ENABLED}
+     * @param localServiceName value of {@link ConfigKeys#TRACING_NAME}
+     *
+     * @return the configured brave tracing
+     */
+    @Produces
+    @Dependent
+    Tracing createTracing(final Reporter<Span> spanReporter,
+                          @ConfigProperty(name = ConfigKeys.TRACING_ENABLED) boolean enabled,
+                          @ConfigProperty(name = ConfigKeys.TRACING_NAME) String localServiceName) {
+
+        final Tracing.Builder builder = Tracing.newBuilder();
+
+        if (!enabled) {
```

Comment:
please stick to the same config as https://github.com/openzipkin/brave-webmvc-example/blob/master/webmvc4/src/main/java/brave/webmvc/TracingConfiguration.java

unnecessary differences make it harder to maintain and harder to understand what's necessarily different vs unnecessarily different. This is just an example codebase, we can create a brave-cdi which has more robust, just  this isn't the place to do that.